File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int product = 1;
        int[] productArr = new int[nums.length];
        for(int i = 0; i < nums.length; i++) {
            for(int j=0;j<nums.length;j++) {
                if(i!=j) {
                    product *= nums[j];
                }
            }
            productArr[i]=product;
            product=1;
        }
        return productArr;
    }
}  

File: Data Structures & Algorithms/subsets-ii/submission-1.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        res = []
        nums.sort()
        def backtrack(nums, path, index):
            res.append(path[:])
            for i in range(index, len(nums)):
                if i > index and nums[i] == nums[i-1]:
                    continue
                path.append(nums[i])
                backtrack(nums, path, i+1)
                path.pop()
            return
        backtrack(nums, [], 0)
        return res